### PR TITLE
feat: improve Eludris rest errors (`ErrorResponse`)

### PR DESCRIPTION
**Pull request by EnokiUN**
## Description

This PR aims to improve how errors in Eludris work to make it easier to work with both server and client wise.

To do so, this PR removes the old (very stupid) way of handling errors and utilises a single enum with some shared data and a macro as to make the code much cleaner.

(also I noticed some quite regarded code whilst refactoring Effis so that also got fixed)

This is related to #34 but does not close it just yet.

## This is a **Code Change**

- [x] Docs have been updated to reflect these changes if necessary. (somewhat, rest will be done along with the other models)
- [x] Changes have been tested.

### CODE REVIEW

File: todel/src/http/response.rs [hunk 1]
```diff
@@ -9,7 +9,12 @@ use crate::models::ErrorResponse;
 
 impl<'r> Responder<'r, 'static> for ErrorResponse {
     fn respond_to(self, req: &'r Request<'_>) -> response::Result<'static> {
-        let status = self.status;
+        let status = match &self {
+            ErrorResponse::RateLimited { shared, .. } => shared.status,
```
Comment (ooliver1) [inline todel/src/http/response.rs:13]: Excuse me?

File: todel/src/models/response.rs [hunk 2]
```diff
@@ -1,121 +1,215 @@
...
+    /// ```
+    Server {
+        #[serde(flatten)]
+        shared: SharedErrorData,
+        /// Extra information about what went wrong.
+        info: String,
+    },
 }
 
+/// Magic marco that handles instantiating [`ErrorResponse`] variants.
```
Comment (ooliver1) [inline todel/src/models/response.rs:105]:
```suggestion
/// Magic marco that handles instantiating [`ErrorResponse`] variants.
```

File: todel/src/models/response.rs [hunk 3]
```diff
@@ -1,121 +1,215 @@
...
+///
+/// It also handles wrapping and returning them when a ratelimiter is passed as the first argument.
```
Comment (ooliver1) [inline todel/src/models/response.rs:107]:
```suggestion
/// It also handles wrapping and returning them when a rate limiter is passed as the first argument.
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: todel/src/http/response.rs (hunk 1 above)
Comment (EnokiUN) [reply todel/src/http/response.rs:13]: I know not of a better way to do it

Comment (ooliver1) [reply todel/src/http/response.rs:13]: ok :neckbeard:

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**